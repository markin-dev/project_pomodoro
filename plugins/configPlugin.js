import * as config from '../constatns';

export default {
    install(Vue) {
        // eslint-disable-next-line
        Vue.prototype.$getConfig = (key) => config[key];
    },
};
