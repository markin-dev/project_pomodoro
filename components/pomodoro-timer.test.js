// изменение констант не должно влиять на тесты

import { shallowMount, createLocalVue } from '@vue/test-utils';
import pomodoroTimer from './pomodoro-timer.vue';
import configPlugin from '../plugins/configPlugin';

describe('pomodoro-timer.vue', () => {
    let localVue;
    const wrapperFactory = () => shallowMount(pomodoroTimer, {
        localVue,
    });

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(configPlugin);
    });

    it('has the expected html structure', () => {
        const wrapper = wrapperFactory();

        expect(wrapper.element).toMatchSnapshot();
    });

    it('correct start time is displayed', () => {
        const wrapper = wrapperFactory();

        const timerText = wrapper.find('.timer__countdown');

        expect(timerText.text()).toBe('00:00');
    });
});
