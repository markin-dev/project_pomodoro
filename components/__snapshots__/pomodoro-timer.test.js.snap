// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pomodoro-timer.vue has the expected html structure 1`] = `
<div>
  <svg
    class="timer"
    height="220"
    width="220"
  >
    <circle
      cx="110"
      cy="110"
      fill="transparent"
      r="105"
      stroke="#424242"
      stroke-width="10"
    />
     
    <circle
      class="timer__progress-circle"
      cx="110"
      cy="110"
      fill="transparent"
      r="105"
      stroke="#42A5F5"
      stroke-width="10"
      style="stroke-dashoffset: 659.7344572538566; stroke-dasharray: 659.7344572538566 659.7344572538566;"
    />
     
    <text
      class="timer__countdown"
      dominant-baseline="middle"
      fill="#42A5F5"
      style="font-size: 59.400000000000006px;"
      text-anchor="middle"
      x="50%"
      y="50%"
    >
      
            00:00
        
    </text>
     
    <circle
      class="timer__clickable-circle"
      cx="110"
      cy="110"
      fill="transparent"
      r="110"
    />
     
    <startbutton-stub
      color="#42A5F5"
      hoverColor="#64B5F6"
      radius="110"
    />
  </svg>
</div>
`;
